### PR TITLE
MINOR: Add a few test cases to clients

### PR DESCRIPTION
**Pull request by see-quick**
This PR adds a few test cases to clients, increasing test coverage a little bit. 

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status

### CODE REVIEW

**Comment by see-quick** [comment]
^ @ijuma @mimaison

File: clients/src/test/java/org/apache/kafka/clients/admin/internals/AbortTransactionHandlerTest.java [hunk 1]
```diff
@@ -52,6 +53,10 @@ public class AbortTransactionHandlerTest {
     @Test
     public void testInvalidBuildRequestCall() {
         AbortTransactionHandler handler = new AbortTransactionHandler(abortSpec, logContext);
+
+        assertEquals(handler.apiName(), "abortTransaction");
+        assertNotNull(handler.lookupStrategy());
```
Comment (mimaison) [inline clients/src/test/java/org/apache/kafka/clients/admin/internals/AbortTransactionHandlerTest.java]: I'm not sure these checks add a lot of value. Especially in this test that is about building invalid calls.

File: clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java [hunk 2]
```diff
@@ -2399,4 +2400,39 @@ public KafkaProducer<T, T> newKafkaProducer() {
         }
     }
 
+    @Test
+    void testSerializerMustNoBeANull() {
```
Comment (mimaison) [inline clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java]: This seems to be the same as `testNoSerializerProvided()`

File: clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java [hunk 3]
```diff
@@ -2399,4 +2400,39 @@ public KafkaProducer<T, T> newKafkaProducer() {
...
+        // delivery.timeout.ms should be equal to or larger than linger.ms + request.timeout.ms
+        assertThrows(KafkaException.class, () -> new KafkaProducer<>(configs, new StringSerializer(), new StringSerializer()));
+
+        configs.put(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, 1000);
+        configs.put(ProducerConfig.LINGER_MS_CONFIG, 999);
+        configs.put(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG, 1);
+
+        assertDoesNotThrow(() -> new KafkaProducer<>(configs, new StringSerializer(), new StringSerializer()));
+
+        configs.put(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, 3000);
```
Comment (mimaison) [inline clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java]: I'm not sure we need this 2nd block as it basically tests the same thing, ie delivery timeout = linger + request timeout

**Review by mimaison — COMMENTED** [review]
Thanks for the PR! I left a few comments.

File: clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java (hunk 2 above)
Comment (see-quick) [reply clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java]: Such test case checks this constructor
```java
public KafkaProducer(Properties properties) {
        this(properties, null, null);
    }
```
but not (so hypothetically if someone change such constructor in the future this could cause problems)
```java
public KafkaProducer(final Map<String, Object> configs) {
        this(configs, null, null);
    }
```
I mean I can remove it if you think that this does not make sense :)

File: clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java (hunk 3 above)
Comment (see-quick) [reply clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java]: This is some leftover...I just wanted to cover two scenarios:
1. \> sign, which throws expectation
```java
configs.put(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, 1000);
configs.put(ProducerConfig.LINGER_MS_CONFIG, 1000);
configs.put(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG, 1);
```
2. and equality sign 
```java
configs.put(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, 1000);
configs.put(ProducerConfig.LINGER_MS_CONFIG, 999);
configs.put(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG, 1);
```
I have modified the code, now it should be correct.

File: clients/src/test/java/org/apache/kafka/clients/admin/internals/AbortTransactionHandlerTest.java (hunk 1 above)
Comment (see-quick) [reply clients/src/test/java/org/apache/kafka/clients/admin/internals/AbortTransactionHandlerTest.java]: Removed

File: clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java [hunk 4]
```diff
@@ -541,6 +541,8 @@ private static int configureDeliveryTimeout(ProducerConfig config, Logger log) {
         int requestTimeoutMs = config.getInt(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG);
         int lingerAndRequestTimeoutMs = (int) Math.min((long) lingerMs + requestTimeoutMs, Integer.MAX_VALUE);
 
+        System.out.println(deliveryTimeoutMs);
```
Comment (mimaison) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java]: Can we undo this debug statement?

File: clients/src/test/java/org/apache/kafka/clients/admin/internals/AbortTransactionHandlerTest.java [hunk 5]
```diff
@@ -52,6 +52,7 @@ public class AbortTransactionHandlerTest {
     @Test
     public void testInvalidBuildRequestCall() {
         AbortTransactionHandler handler = new AbortTransactionHandler(abortSpec, logContext);
+
```
Comment (mimaison) [inline clients/src/test/java/org/apache/kafka/clients/admin/internals/AbortTransactionHandlerTest.java]: Can we undo this unrelated change?

File: clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java (hunk 2 above)
Comment (mimaison) [reply clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java]: Right, it's not exactly the same. If you really want to keep it, maybe we can just add this check to the existing test?

**Review by mimaison — COMMENTED** [review]
Thanks for the updates. There's a couple of leftovers to remove, otherwise it should be good to go.

File: clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java (hunk 2 above)
Comment (see-quick) [reply clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java]: Okay, I have moved the check to `testNoSerializerProvided`

**Review by mimaison — APPROVED** [review]
LGTM , thanks for the PR!
